File: app/backend/java/auth-service/src/main/java/com/srbms/authservice/entity/Complaint.java
package com.srbms.authservice.entity;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "complaint")
public class Complaint {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "complaint_id")
    private Integer complaintId;

    @Column(name = "user_id", nullable = false)
    private Integer userId;

    @Column(name = "ride_id")
    private Integer rideId;

    @Column(name = "subject", nullable = false, length = 150)
    private String subject;

    @Column(name = "description", nullable = false, columnDefinition = "TEXT")
    private String description;

    @Column(name = "category", length = 50)
    private String category = "General";

    @Column(name = "status")
    private String status = "Open";

    @Column(name = "resolution_notes", columnDefinition = "TEXT")
    private String resolutionNotes;

    @Column(name = "created_at", insertable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", insertable = false, updatable = false)
    private LocalDateTime updatedAt;

    public Complaint() {}

    public Complaint(Integer userId, Integer rideId, String subject, String description, String category) {
        this.userId = userId;
        this.rideId = rideId;
        this.subject = subject;
        this.description = description;
        this.category = category;
        this.status = "Open";
    }

    public Integer getComplaintId() {
        return complaintId;
    }

    public void setComplaintId(Integer complaintId) {
        this.complaintId = complaintId;
    }

    public Integer getUserId() {
        return userId;
    }

    public void setUserId(Integer userId) {
        this.userId = userId;
    }

    public Integer getRideId() {
        return rideId;
    }

    public void setRideId(Integer rideId) {
        this.rideId = rideId;
    }

    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getResolutionNotes() {
        return resolutionNotes;
    }

    public void setResolutionNotes(String resolutionNotes) {
        this.resolutionNotes = resolutionNotes;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }
}

File: app/backend/.net/SmartRide.AuthService/Models/Entities/Complaint.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SmartRide.AuthService.Models.Entities;

[Table("complaint")]
public class Complaint
{
    [Key]
    [Column("complaint_id")]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int ComplaintId { get; set; }

    [Required]
    [Column("user_id")]
    public int UserId { get; set; }

    [Column("ride_id")]
    public int? RideId { get; set; }

    [Required]
    [Column("subject")]
    [StringLength(150)]
    public string Subject { get; set; } = string.Empty;

    [Required]
    [Column("description", TypeName = "TEXT")]
    public string Description { get; set; } = string.Empty;

    [Column("category")]
    [StringLength(50)]
    public string Category { get; set; } = "General";

    [Column("status")]
    [StringLength(50)]
    public string Status { get; set; } = "Open";

    [Column("resolution_notes", TypeName = "TEXT")]
    public string? ResolutionNotes { get; set; }

    [Column("created_at")]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("updated_at")]
    [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public Complaint() { }

    public Complaint(int userId, int? rideId, string subject, string description, string category)
    {
        UserId = userId;
        RideId = rideId;
        Subject = subject;
        Description = description;
        Category = category;
        Status = "Open";
    }
}
